monai/apps/deepedit/transforms.py

```diff
@@ -95,25 +95,22 @@ def __init__(self, keys: KeysCollection, label_names=None, allow_missing_keys: b
 
         self.label_names = label_names
 
-    def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
         d: Dict = dict(data)
         for key in self.key_iterator(d):
-            if key == "label":
-                # Dictionary containing new label numbers
-                new_label_names = {}
-                label = np.zeros(d[key].shape)
-                # Making sure the range values and number of labels are the same
-                for idx, (key_label, val_label) in enumerate(self.label_names.items(), start=1):
-                    if key_label != "background":
-                        new_label_names[key_label] = idx
-                        label[d[key] == val_label] = idx
-                    if key_label == "background":
-                        new_label_names["background"] = 0
+            # Dictionary containing new label numbers
+            new_label_names = {}
+            label = torch.zeros_like(d[key])
```

Comment:
I think you can still use `label = np.zeros(d[key].shape)` here

Reply:
this causes errors...https://github.com/Project-MONAI/MONAI/runs/7746818049?check_suite_focus=true
```
======================================================================
ERROR: test_correct_results_0 (tests.test_deepedit_transforms.TestNormalizeLabelsDatasetd)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/parameterized/parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/tmp/tmp.vk3Zowp95Y/tests/test_deepedit_transforms.py", line 272, in test_correct_results
    result = add_fn(input_data)
  File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/monai/apps/deepedit/transforms.py", line 103, in __call__
    label = torch.zeros_like(d[key])
TypeError: zeros_like(): argument 'input' (position 1) must be Tensor, not numpy.ndarray

----------------------------------------------------------------------
```

Reply:
Yes, I don't get what's the issue here. Before when I had this: 

> def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:

It also shows this error :(

Reply:
Just changed to numpy array  :)

Reply:
looks great, thank you!